R/calc_Huntley2006.R

```diff
@@ -540,17 +546,48 @@ calc_Huntley2006 <- function(
 
     if (!inherits(fit_sim, "try-error"))
       coefs <- coef(fit_sim)
-    else
-      .throw_error("Could not fit simulated curve, check suitability of ",
-                   "model and parameters")
+    else {
+      ## As the fit from the given starting values failed, try again by fixing
+      ## D0 at different values to make the fitting a bit easier. Values are
+      ## spaced logarithmically so that the ratio between consecutive values
+      ## is constant.
+      all.D0 <- c(start$D0, exp(seq(log(start$D0 / 10), log(start$D0 * 10),
+                                    length.out = 99)))
+      fit.D0 <- lapply(1:length(all.D0), function(idx) {
+        D0 <- all.D0[idx]
+        t <- try(minpack.lm::nlsLM(
+                  formula = model,
+                  start = start[-2],
+                  lower = lower.bounds[-2],
+                  upper = upper.bounds[-2],
+                  control = list(maxiter = settings$maxiter)),
+                 silent = TRUE)
+
+        if (inherits(t, "try-error"))
+          return(NULL)
+        return(t)
+      })
```

Comment:
I think we should I think we should parallelise this part to let it run faster on multicore machines.